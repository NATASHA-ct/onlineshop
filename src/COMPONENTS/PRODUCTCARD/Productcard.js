/* eslint-disable */
import { Link } from "react-router-dom";

const Productcard= (props) => {
  const { id, title, price, image} = props;
  return (
    <Link to={`/product/${id}`}>
      <div className="card" id={id}>
        <div className="card-container">
          <img className="card-img" src={image} width="200px" height="200px" />
          <div className="card-details">
            <h1 className="card-title">{title}</h1>
            <p className="card-price">$ {price}</p>
          </div>
          
       </div>
      </div>
    </Link>
  );
};

export default Productcard;
