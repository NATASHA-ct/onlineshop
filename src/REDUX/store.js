/* eslint-disable */
import { configureStore } from "@reduxjs/toolkit";
import productReducer from "./products/productsSlice";
import productDetailsReducer from "./products/productsdetailsSlice";

const store = configureStore({
  reducer: {
    products: productReducer,
    productDetail: productDetailsReducer,
  },
});

export default store;
