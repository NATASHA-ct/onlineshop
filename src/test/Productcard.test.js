import { BrowserRouter } from "react-router-dom";
import store from "../REDUX/store";
import Productcard from "../COMPONENTS/PRODUCTCARD/Productcard";
import { Provider } from "react-redux";
import renderer from 'react-test-renderer';

describe("to test the Productcard components", () => {
  test("matches the Productcard DOM snapshot", () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Productcard />
        </Provider>
      </BrowserRouter>
    );
    expect(tree).toMatchSnapshot();
  });
});
