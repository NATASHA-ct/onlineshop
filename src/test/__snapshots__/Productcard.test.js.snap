// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to test the Productcard components matches the Productcard DOM snapshot 1`] = `
<a
  href="/product/undefined"
  onClick={[Function]}
>
  <div
    className="card"
  >
    <div
      className="card-container"
    >
      <img
        className="card-img"
        height="200px"
        width="200px"
      />
      <div
        className="card-details"
      >
        <h1
          className="card-title"
        />
        <p
          className="card-price"
        >
          $ 
        </p>
      </div>
    </div>
  </div>
</a>
`;
